---
layout: draft
title: 每周工作总结：2016年2月22日
date: 2016-02-22
comments: true
category: Weekly Reflection
tags: reflection
keywords: Weekly Reflection

---

## 2016年02月26日

今天在测试系统的时候碰到了rsync的一个问题：无论是使用Linux命令行还是在Python中通过subprocess.call()进行调用，源文件和目标文件都可以正确地被一致拷贝，也就是说，拷贝完毕之后源文件和目标文件没有任何差别。但是当我使用真实环境的时候，发现拷贝完毕的目标文件夹中所有子文件夹的size全都不对，导致在进行integrity check的时候总是失败。

反复测试之后，我终于发现了问题的原因：在生产环境中，源文件存储在Linux ext4文件系统中，而拷贝的目标地址却是在Windows NTFS文件系统中。由于两个文件系统对目录的处理方式不一样，导致二者的size无法维持一致。而我在使用rsync拷贝和做一致性检查的时候，使用了"-a"参数，需要保证二者尽量一致，也就是一致性检查比较严格，所以导致检查总是失败。一种可能的解决办法是：只使用"-r"进行拷贝和一致性检查。

反思：测试环境一定要最大程度地模拟生产环境！否则在测试环境中运行良好的代码在生产环境中会产生问题！
